### ISSUE TITLE
With version 1.5.0 images are no longer deleted

### ISSUE THREAD

**Issue by eickehnn** [snok/container-retention-policy#41]
The following still worked up to version 1.4.2, with version 1.5.0 images are no longer deleted:
```yaml
  delete-older-images:
    name: Delete older ghcr images
    runs-on: ubuntu-latest
    steps:
      - name: Delete images
        uses: snok/container-retention-policy@v1
        with:
          image-names: repo/imagename
          cut-off: now UTC+1
          account-type: org
          org-name: my-org
          keep-at-least: 2
          skip-tags: latest
          token: ${{ secrets.PAT }}
```
Note: There is no log output or error message

**Comment by sondrelg** [comment]
Huh, that's unfortunate. Are your workflow runs private, or could you link me one?

These are all the changes since v1.4.2: https://github.com/snok/container-retention-policy/compare/main..3e6a692f35a6d52db6fde535877fe2fd379dfcb4

**Comment by eickehnn** [comment]
I made a test example: https://github.com/eickehnn/container-retention-policy-test

**Comment by sondrelg** [comment]
That's really helpful, thanks!

Looking at the [changes](https://github.com/snok/container-retention-policy/compare/3e6a692f35a6d52db6fde535877fe2fd379dfcb4..main) from 1.4.2 again I wonder if there might be a problem with `main.py` line 409, or something like that. 

My own projects seem to work fine on v1.5.0, and I'm *guessing* the only relevant difference between the two is that your image name contains a `/` while mine don't.

I won't have time to look at this properly for at least a few days, but if you have time, I'd consider calling the Github API directly to see what data you're receiving and whether the `fnmatch` logic needs to be adjusted to account for special characters. If you figure anything out, a PR is of course very welcome 🙂 

The docs can be found [here](https://docs.github.com/en/rest/packages#list-packages-for-a-user) and [here](https://github.com/snok/container-retention-policy/blob/eeb1249ebb557c2f7ff1de16f46282a82791017f/main.py#L90) is the relevant code from the action 👍

**Comment by lukasz-mitka** [comment]
@eickehnn how many packages do you have?
Try using `@main` it contains #44 which was a workaround for me.
Full fix will be done in #42.

**Comment by eickehnn** [comment]
In the current example, it's one image with currently four versions that should be reduced to two. I try it with main and the both latest commits from pr42 and pr44, but the issues is still there.

**Comment by lukasz-mitka** [comment]
I didn't read `sondrelg` comment carefully before, he is definitely right. Things I have mentioned won't work in your case, sorry.

**Comment by den-is** [comment]
In my case `/` doesn't interfere with the action.
Tested this action python code locally. version and images show up just fine.
In my case, the problem is that I have more than 100 images, and images that require deleting are "deep" in the list.
Waiting for this PR to be merged: https://github.com/snok/container-retention-policy/pull/42/files

**Comment by andreykaipov** [comment]
> I made a test example: https://github.com/eickehnn/container-retention-policy-test

I think the issue in the example is it's using `${{ github.repository }}` to search for the image names. But this variable actually expands to `owner/repo`.
 
This was my issue too, since I had wrongly assumed my username was part of the image name. But after checking the API, the name was... you know, the actual name:

```console
❯ gh api /users/andreykaipov/packages?package_type=container | jq -r .[].name
goobs
```

In hindsight that makes more sense than including the username. I ended up using `${{ github.event.repository.name }}` as the image names to search for, so I could parametrize the policy to the repo name my workflow is in (since my image name is just the repo name).

Since the example [pushes to `ghcr.io/${{ github.repository }}/alpine-test`](https://github.com/eickehnn/container-retention-policy-test/blob/main/.github/workflows/build-image.yml#L33), the image names to search for should actually be `${{ github.event.repository.name }}/alpine-test`, where `${{ github.event.repository.name }}` is the `repo` in `owner/repo`.

**Comment by eickehnn** [comment]
Change the variables from ${{ github.repository }} to ${{ github.event.repository.name }} or replacing them with absolute values makes no difference, the images in this example are not deleted with version 1.5.0, but with version 1.4.2 they are.

**Comment by fabricio-bastian** [comment]
I have the same issue with 1.5.0. Is this issue fixed in 1.5.1?

**Comment by sondrelg** [comment]
The suggestions above seem to indicate that it's caused by an image name mismatch. I'll close this for now, under that assumption, since we've not otherwise been able to replicate :slightly_smiling_face: Feel free to reopen if anyone has new information
